### PR TITLE
SemVer2 - Missing db index on Packages table #498

### PR DESCRIPTION
**Pull request by xavierdecoster**
Noticed an increase in DTU consumption on the gallery db after deploying semver2, and it's caused by a missing index on the Packages table.

The Query Processor estimates that implementing the following index could improve the query cost by 99.5235%.

Affected EF query:

```sql
(@p__linq__0 int, @p__linq__4 int,@p__linq__5 int)

SELECT	[GroupBy1].[A1] AS [C1]
FROM 
( 
	SELECT	COUNT(1) AS [A1]
    FROM	[dbo].[Packages] AS [Extent1]
    WHERE	([Extent1].[Deleted] <> 1) 
		AND (([Extent1].[SemVerLevelKey] = @p__linq__0) 
			OR (([Extent1].[SemVerLevelKey] IS NULL) AND (@p__linq__0 IS NULL))
			) 
		AND ((
			CASE 
			WHEN (@p__linq__4 = @p__linq__5) 
			THEN [Extent1].[IsLatestSemVer2] 
			ELSE [Extent1].[IsLatest] 
			END) = 1)
)  AS [GroupBy1]
```

### CODE REVIEW

**Comment by joelverhagen** [comment]
What are the parameters in this case? What code path does this primarily affect?

**Comment by xavierdecoster** [comment]
@joelverhagen https://github.com/NuGet/NuGetGallery/blob/master/src/NuGetGallery/Controllers/ODataV2FeedController.cs#L86

File: src/NuGetGallery/Migrations/201706080930506_AddIndexSemVerLevelKeyPackageRegistrationKey.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace NuGetGallery.Migrations
+{
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class AddIndexSemVerLevelKeyPackageRegistrationKey : DbMigration
+    {
+        public override void Up()
+        {
+            Sql("IF NOT EXISTS(SELECT * FROM sys.indexes WHERE name = 'nci_wi_Packages_SemVerLevelKey_PackageRegistrationKey' AND object_id = OBJECT_ID('Packages')) CREATE NONCLUSTERED INDEX [nci_wi_Packages_SemVerLevelKey_PackageRegistrationKey] ON [dbo].[Packages] ([SemVerLevelKey],[IsPrerelease],[Deleted]) INCLUDE ([PackageRegistrationKey],[Description],[Tags])");
```
Comment (scottbommarito) [inline src/NuGetGallery/Migrations/201706080930506_AddIndexSemVerLevelKeyPackageRegistrationKey.cs]: Any reason for these not to be in a single migration?

Comment (xavierdecoster) [inline src/NuGetGallery/Migrations/201706080930506_AddIndexSemVerLevelKeyPackageRegistrationKey.cs]: Yes, they were separate PRs, and have been evaluated separately to find out whether the change was for the better or not.

**Comment by xavierdecoster** [comment]
Want to point out these indices have already been applied (see offline discussion), and monitoring has clearly shown they had a positive effect on gallery db DTU consumption.

**Pull request merged**